azure/cloud/Redis.rego

```diff
@@ -285,4 +285,137 @@ arc_private_endpoint_metadata := {
     "Resource Type": "microsoft.cache/redis",
     "Policy Help URL": "",
     "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis"
+}
+
+
+#
+# PR-AZR-CLD-ARC-007
+
+default min_tls_version_redis = null
+
+azure_attribute_absence["min_tls_version_redis"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    not resource.properties.minimumTlsVersion
+}
+
+azure_issue["min_tls_version_redis"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    to_number(resource.properties.minimumTlsVersion) != 1.2
+}
+
+min_tls_version_redis {
+	resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis"
+    not azure_attribute_absence["min_tls_version_redis"]
+    not azure_issue["min_tls_version_redis"]
+}
+
+
+min_tls_version_redis = false {
+    azure_issue["min_tls_version_redis"]
+}
+
+min_tls_version_redis = false {
+    azure_attribute_absence["min_tls_version_redis"]
+}
+
+min_tls_version_redis_err = "Azure Redis Cache currently doesn't have latest version of tls configured" {
+    azure_issue["min_tls_version_redis"]
+} else = "Microsoft.Cache/redis property 'minimumTlsVersion' need to be exist. Its missing from the resource. Please set the value to '1.2' after property addition." {
+    azure_attribute_absence["min_tls_version_redis"]
+}
+
+min_tls_version_redis_metadata := {
+    "Policy Code": "PR-AZR-CLD-ARC-007",
+    "Type": "Cloud",  
+    "Product": "AZR",
+    "Language": "",
+    "Policy Title": "Ensure Azure Redis Cache has latest version of tls configured",
+    "Policy Description": "This policy will identify the Azure Redis Cache which doesn't have the latest version of tls configured and give the alert",
+    "Resource Type": "microsoft.cache/redis",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://docs.microsoft.com/en-us/azure/templates/microsoft.cache/redis"
+}
+
+
+
+#
+# PR-AZR-CLD-ARC-008
+#
+
+default redis_cache_firewall_not_allowing_full_inbound_access = null
+
+
+azure_attribute_absence ["redis_cache_firewall_not_allowing_full_inbound_access"] {
+    count([c | lower(input.resources[_].type) == "microsoft.cache/redis/firewallrules"; c := 1]) == 0
+}
+
+
+azure_attribute_absence["redis_cache_firewall_not_allowing_full_inbound_access"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis/firewallrules"
+    not resource.properties.startIP
+}
+
+
+azure_attribute_absence["redis_cache_firewall_not_allowing_full_inbound_access"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.cache/redis/firewallrules"
+    not resource.properties.endIP
+}
+
+
+azure_issue["redis_cache_firewall_not_allowing_full_inbound_access"] {
```

Comment:
@shahinshirvani use dependsOn check in case of ARM and cloud.

See PR-AZR-ARM-SQL-031 for example.